packages/enketo-express/public/js/src/module/translator.js

```diff
@@ -89,32 +89,39 @@ const localize = (container, lng) => {
             list.forEach((el) => {
                 const key = el.dataset.i18n;
                 if (key) {
-                    if (!cache[key]) {
-                        let options = {};
-                        // quick hack for __icon__ replacement
-                        if (el.dataset.i18nIcon) {
-                            options = {
-                                icon: `<span class="icon ${el.dataset.i18nIcon}"> </span>`,
-                                interpolation: { escapeValue: false },
-                            };
-                        }
-                        if (el.dataset.i18nNumber) {
-                            options = {
-                                number: el.dataset.i18nNumber,
-                            };
-                        }
-                        cache[key] = t(key, options);
+                    let options = {};
+                    // quick hack for __icon__ replacement
+                    if (el.dataset.i18nIcon) {
+                        options = {
+                            icon: `<span class="icon ${el.dataset.i18nIcon}"> </span>`,
+                            interpolation: { escapeValue: false },
+                        };
                     }
+                    if (el.dataset.i18nNumber) {
+                        options = {
+                            number: el.dataset.i18nNumber,
+                        };
+                    }
+
+                    // Update cache only if the translation is new or has changed
+                    const translationText = t(key, options);
```

Comment:
If you're always generating a translation, does the translation cache serve any purpose? Presumably, the translation cache is meant to speed up loading time or language-switching time for a translated form. 

If the cache serves no purpose any more after your fix, you could either:
1. remove it after investigating that for a very long translated form the cache makes no meaningful improvement on loading
2. bypass the cache if there is an i18number (or something more intelligent than that)
3. include the number in the cache key

Reply:
Hi @MartijnR, new commit addressing the issue with solution that is similar to your suggestion number 3.
I setup cache key as i18n-i18nNumber.

Thanks,